src/main/java/org/apache/sling/bundleresource/impl/BundleResource.java

```diff
@@ -229,7 +228,11 @@ public String getPath() {
 
     @Override
     public String getResourceType() {
-        return this.valueMap.get(ResourceResolver.PROPERTY_RESOURCE_TYPE, String.class);
+        String resourceType = this.valueMap.get(ResourceResolver.PROPERTY_RESOURCE_TYPE, String.class);
+        if ( resourceType == null ) {
+            resourceType = this.isFolder ? NT_FOLDER : NT_FILE;
```

Comment:
food for thoughts: Why do we use JCR nodetypes here? Wouldn't it be possible to use some sling "default resourcetypes"?

Reply:
I've checked DefaultGetServlets and it looks like this type is not used by default for rendering the content. For directories, we fall into this condition: https://github.com/apache/sling-org-apache-sling-servlets-get/blob/org.apache.sling.servlets.get-2.1.44/src/main/java/org/apache/sling/servlets/get/impl/helpers/StreamRenderer.java#L131 , for files we pass control back to BundleResource: https://github.com/apache/sling-org-apache-sling-servlets-get/blob/org.apache.sling.servlets.get-2.1.44/src/main/java/org/apache/sling/servlets/get/impl/helpers/StreamRenderer.java#L157

Reply:
@joerghoh are you thinking about default resource types across all resource providers? Lets say: `sling/default/file`, `sling/deafult/folder` or `sling/default/unstructure`? 

Or rather to introduce something specific to BundleResource, like we have it in ServletResource superType [here](https://github.com/apache/sling-org-apache-sling-servlets-resolver/blob/0e3cdecc9cf4f88a740ba5fc86dec8d1dc9e2c17/src/main/java/org/apache/sling/servlets/resolver/internal/resource/ServletResource.java#L40)?

Reply:
The following implementation (local default resource types) seems to be working:
```
    @Override
    public String getResourceType() {
        String resourceType = this.valueMap.get(ResourceResolver.PROPERTY_RESOURCE_TYPE, String.class);
        if ( resourceType == null ) {
            resourceType = this.isFolder ? "sling/bundle/folder" : "sling/bundle/file";
        }
        return resourceType;
    }
```

Reply:
For me using completely JCR agnostic resource types should be tracked in a follow-up PR. IMHO this PR is only about removing the resource type from the valueMap.

Reply:
Agree, out of the scope of this PR.